### DISCUSSION TITLE
Attach hangouts meet conference data with event

### DISCUSSION THREAD

**Discussion by Sohail-Ahmad-PRO** [spatie/laravel-google-calendar#203] [Q&A]
How can I attach google hangouts/meet conference data/link to a google calendar event in this package. Could you please let me know if this package has such support?

**Comment by emanuelbiet** [comment]
Did you achieve any progress on this issue?

**Reply by Sohail-Ahmad-PRO** [reply]
@emanuelbiet Not yet. There should be support for scheduling google meet meetings and get its conference details in the event object but currently, it's not happening.

**Comment by tomcodes** [comment]
Google Calendar API specifies that it is possible:  
https://developers.google.com/calendar/api/v3/reference/events#conferenceData

You can always work on it and submit a PR to this package to make it happen.

**Comment by StefanoDucciConvenia** [comment]
Hello @emanuelbiet  @Sohail-Ahmad-PRO , you can set conferenceData property of your event like this:

    $event = new Event;
    $event->name = 'A new event';
    $event->description = 'Event description';
    $event->startDateTime = Carbon\Carbon::now();
    $event->endDateTime = Carbon\Carbon::now()->addHour();
    $event->conferenceData = [
        'createRequest' => [
            'conferenceSolutionKey' => [
                'type' => 'hangoutsMeet'
            ],
            'requestId' => 'sdg-mjae-hat'
        ]
    ];
    $event->addAttendee([
        'email' => 'john.doe@email.com',
        'name' => 'John Doe',
        'comment' => 'Lorum ipsum',
    ]);
    $event->save('insertEvent', [
        'sendUpdates' => 'all',
        'conferenceDataVersion' => 1,
    ]);

**Reply by berkaykaradeniz** [reply]
is this working ? @StefanoDucciConvenia

**Reply by reddigriz** [reply]
@berkaykaradeniz 

I am testing it with a GSUITE using a service account, it works correctly, but a function that handles the requestId code generation is needed

Usually the Google endpoint handles the code generation if none is presented, but somehow it's not working for me, so I just wrote it

**Reply by StefanoDucciConvenia** [reply]
Yes exactly  it is working. But you need to send a random requestId, even though this id is not the final id of the meeting.

You can get the real id from returned event of save method
